shaker/salt_shaker.py

```diff
@@ -420,7 +420,7 @@ def _rev_to_sha(self, formula, repo):
             have_updated = True
 
 
-def get_formulas(root_formula=None, root_dir='.', constraint=None):
+def get_formulas(root_formula=None, root_dir='.', constraint=None, formula_exclusions=[]):
```

Comment:
Python default arguments should never be objects because if you call `get_formulas` twice in the same interpreter, the second time around you will reuse *the same array as last time* - you won't get a new array again.

The way to deal with this is something like this:

```python
    if formula_exclusions is None:
        formula_exclusions = []
```